### ISSUE TITLE
How to run AFL?

### ISSUE THREAD

**Issue by frankmorgner** [google/oss-fuzz#2094]
I have set up a working Dockerfile and build script for OSS-Fuzz and I have a working script to test OpenSC with AFL locally, which looks like this:
```sh
#!/bin/bash

if [ ! -d x41-smartcard-fuzzing ];
then
    git clone https://github.com/x41sec/x41-smartcard-fuzzing
fi

gcc -shared -fPIC -o x41-smartcard-fuzzing/scard_override/libsccard_override.so x41-smartcard-fuzzing/scard_override/scard_override.c -ldl -I/usr/include/PCSC/

if [ ! -f configure ];
then
    autoreconf -vis
fi

if [ ! -f Makefile ];
then
    make clean
fi

#export AFL_USE_ASAN=1
./configure CC=afl-gcc CFLAGS="-O0" LDFLAGS="-ldl" --disable-shared --disable-notify --with-pcsc-provider=$PWD/x41-smartcard-fuzzing/scard_override/libsccard_override.so
make

if [ ! -d tests/fuzzing-testcases ]
then
    mkdir -p tests/fuzzing-testcases
    echo -ne "$(printf '\\x90\\x00')" > tests/fuzzing-testcases/9000
    IN=tests/fuzzing-testcases
else
    IN=-
fi

FUZZ_FILE=input.apdu  afl-fuzz -i $IN -o out -f input.apdu src/tools/pkcs15-tool -D
```
However, I don't know how to integrate this test script with OSS-Fuzz. While there are many examples that use libFuzzer, and AFL *is* supported, I couldn't find any project that uses AFL in a similar way, that I planned...

**Comment by jonathanmetzman** [comment]
OSS-Fuzz supports using AFL for fuzzing libFuzzer style targets.
This means it uses clang for instrumentation and a [shim](https://cs.chromium.org/chromium/src/third_party/libFuzzer/src/afl/afl_driver.cpp?q=afl_driver.c&sq=package:chromium&g=0&l=1) to fuzz `LLVMFuzzerTestOneInput` functions (using AFL's persistent mode).
OSS-Fuzz doesn't have support for fuzzing traditional AFL targets compiled with afl-{gcc,clang}
Basically, you need to set up a libFuzzer target to build [like this](https://github.com/google/oss-fuzz/blob/master/docs/new_project_guide.md#buildsh) and OSS-Fuzz will be able to build it with AFL.

**Comment by frankmorgner** [comment]
I see, thanks
